src/OpenLoco/src/Date.cpp

```diff
@@ -92,54 +92,52 @@ namespace OpenLoco
     }
 
     // 0x00495C65
-    // eax: totalDays
-    // returns:
-    //    eax: year
-    //    ebx: month
-    //    edx: day
-    Date calcDate(const uint32_t totalDays)
+    Date calcDate(uint32_t totalDays)
     {
         constexpr auto kBaseYear = 1800;
-        constexpr auto kDaysInYear = 365;
-        constexpr auto kDaysInOlympiad = (365 * 4) + 1; // Useful because the calendar that Loco uses (Julian) has a 4-year cycle.
-        constexpr auto kFeb29 = 31 + 28;
-
-        int32_t years = ((totalDays / kDaysInOlympiad) & 0xFFFF) * 4;
-        int32_t day = totalDays % kDaysInOlympiad;
-
-        // Count the years and add an extra day for when there isn't leap year
-        if (day > kDaysInYear)
+        auto y = kBaseYear;
+        while (totalDays > 365)
```

Comment:
this isn't that efficient, if anyone can come up with a closed form solution that'd be cool